Userland/Utilities/test-fuzz.cpp

```diff
@@ -0,0 +1,130 @@
+ * modification, are permitted provided that the following conditions are met:
+ *
+ * 1. Redistributions of source code must retain the above copyright notice, this
+ *    list of conditions and the following disclaimer.
+ *
+ * 2. Redistributions in binary form must reproduce the above copyright notice,
+ *    this list of conditions and the following disclaimer in the documentation
+ *    and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE
+ * FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+ * DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+ * OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#include <AK/String.h>
+#include <LibCore/ArgsParser.h>
+#include <LibCore/File.h>
+#include <stdio.h>
+#include <unistd.h>
+
+#define ENUMERATE_TARGETS(T) \
+    T(BMPLoader)             \
+    T(ELF)                   \
+    T(Gemini)                \
+    T(GIFLoader)             \
+    T(HttpRequest)           \
+    T(ICOLoader)
+/*
+//    T(JPGLoader) \
+//    T(Js) \
+//    T(Markdown) \
+//    T(PBMLoader) \
+//    T(PGMLoader) \
+//    T(PNGLoader) \
+//    T(PPMLoader) \
+//    T(RegexECMA262) \
+//    T(RegexPosixExtended) \
+//    T(Shell) \
+//    T(TTF) \
+//    T(URL)
+*/
+
+#undef __ENUMERATE_TARGET
+#define __ENUMERATE_TARGET(x) extern "C" int Test##x(const uint8_t*, size_t);
+ENUMERATE_TARGETS(__ENUMERATE_TARGET)
+#undef __ENUMERATE_TARGET
+
+// FIXME: How to automate this?
+#define LLVMFuzzerTestOneInput TestBMPLoader
+#include <Meta/Lagom/Fuzzers/FuzzBMPLoader.cpp>
+#undef LLVMFuzzerTestOneInput
+
+#define LLVMFuzzerTestOneInput TestELF
+#include <Meta/Lagom/Fuzzers/FuzzELF.cpp>
+#undef LLVMFuzzerTestOneInput
+
+#define LLVMFuzzerTestOneInput TestGemini
+#include <Meta/Lagom/Fuzzers/FuzzGemini.cpp>
+#undef LLVMFuzzerTestOneInput
+
+#define LLVMFuzzerTestOneInput TestGIFLoader
+#include <Meta/Lagom/Fuzzers/FuzzGIFLoader.cpp>
+#undef LLVMFuzzerTestOneInput
+
+#define LLVMFuzzerTestOneInput TestHttpRequest
+#include <Meta/Lagom/Fuzzers/FuzzHttpRequest.cpp>
+#undef LLVMFuzzerTestOneInput
+
+#define LLVMFuzzerTestOneInput TestICOLoader
+#include <Meta/Lagom/Fuzzers/FuzzICOLoader.cpp>
+#undef LLVMFuzzerTestOneInput
+
+typedef int (*TestOneInputFn)(const uint8_t*, size_t);
+
+static TestOneInputFn parse_target_name(const String& name)
```

Comment:
`auto` should be able to infer this, I think.
Below too, so you can very likely omit the typedef above

Reply:
I didn't know that `auto` works for return types!